toolbox/util/githubClient.go

```diff
@@ -244,28 +244,41 @@ func (g GithubClient) CloseIdlePullRequests(prTitlePrefix, repo, baseBranch stri
 
 // GetHeadCommitSHA finds the SHA of the commit to which the HEAD of branch points
 func (g GithubClient) GetHeadCommitSHA(repo, branch string) (string, error) {
-	return g.getReferenceSHA(repo, "refs/heads/"+branch)
+	sha, _, err := g.GetReference(repo, "refs/heads/"+branch)
+	return sha, err
 }
 
 // GetTagCommitSHA finds the SHA of the commit from which the tag was made
 func (g GithubClient) GetTagCommitSHA(repo, tag string) (string, error) {
-	sha, err := g.getReferenceSHA(repo, "refs/tags/"+tag)
+	sha, ty, err := g.GetReference(repo, "refs/tags/"+tag)
 	if err != nil {
 		return "", err
 	}
-	tagObj, _, err := g.client.Git.GetTag(
-		context.Background(), g.owner, repo, sha)
-	if err != nil {
-		return "", err
+
+	if ty == "tag" {
```

Comment:
Change here because some tags point to a tag while some point to a commit
```Bash
 GET https://api.github.com/repos/istio/istio/git/refs/tags/0.2.2
{
    "ref": "refs/tags/0.2.2",
    "url": "https://api.github.com/repos/istio/istio/git/refs/tags/0.2.2",
    "object": {
        "sha": "9ac006e070b88dba19ad26048c1010675d77e321",
        "type": "commit",
        "url": "https://api.github.com/repos/istio/istio/git/commits/9ac006e070b88dba19ad26048c1010675d77e321"
    }
}
```
```Bash
GET https://api.github.com/repos/istio/mixer/git/refs/tags/0.2.2
{
    "ref": "refs/tags/0.2.2",
    "url": "https://api.github.com/repos/istio/mixer/git/refs/tags/0.2.2",
    "object": {
        "sha": "f146a9ff912337c3d17261140d69801bd6c5789b",
        "type": "tag",
        "url": "https://api.github.com/repos/istio/mixer/git/tags/f146a9ff912337c3d17261140d69801bd6c5789b"
    }
}
```